### PR TITLE
✨Remove FIXME (adjust the comment and code both)

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

1) remove one FIXME as it's expected
2) remove another FIXME as we don't need get port first then delete it, we already have handle for 404 error
**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

**Review by apricote — APPROVED** [review]

**Review by mdbooth — COMMENTED** [review]
Nice cleanup! I think the other FIXME should remain, though, as that issue is not resolved: we're still doing an unnecessary lookup for a trunk which we should already know doesn't exist.

I'm pretty sure we'll fix it anyway, so I don't feel too strongly about it, but my preference would be it stays.

/lgtm

File: pkg/cloud/services/compute/instance_test.go [hunk 1]
```diff
@@ -1190,10 +1188,8 @@ func TestService_DeleteInstance(t *testing.T) {
 					},
 				}}, nil)
 				computeRecorder.DeleteAttachedInterface(instanceUUID, portUUID).Return(nil)
-				// FIXME: Why we are looking for a trunk when we know the port is not trunked?
+				// ListTrunk will be called as we have trunk extension but no trunk associated with the port
```
[the remark this answers is not in the document]
Comment (jichenjc) [reply pkg/cloud/services/compute/instance_test.go:1191]: will update and give another thought , thanks :)

**Comment by jichenjc** [comment]
@mdbooth updated, please take another look :) 

/hold cancel

**Pull request merged**